front_end/ndb/NdbMain.js

```diff
@@ -152,34 +138,29 @@ Ndb.ContextMenuProvider = class {
 };
 
 Ndb.NodeProcessManager = class extends Common.Object {
-  constructor(targetManager, service) {
+  constructor(targetManager) {
     super();
-    this._service = service;
-    this._processes = new Map();
+    this._service = null;
     this._lastDebugId = 0;
     this._lastStarted = null;
     this._targetManager = targetManager;
     this._cwds = new Map();
-    this._isProfiling = false;
+    this._finishProfiling = null;
     this._cpuProfiles = [];
     this._targetManager.addModelListener(
         SDK.RuntimeModel, SDK.RuntimeModel.Events.ExecutionContextDestroyed, this._onExecutionContextDestroyed, this);
-    InspectorFrontendHost.sendMessageToBackend = this._sendMesage.bind(this);
   }
 
   static async create(targetManager) {
-    const service = await Ndb.backend.createService('ndd_service.js');
-    const manager = new Ndb.NodeProcessManager(targetManager, service);
-    await service.init(rpc.handle(manager));
+    const manager = new Ndb.NodeProcessManager(targetManager);
+    const service = await Ndb.backend.createService('ndd_service.js', rpc.handle(manager));
+    manager._service = service;
```

Comment:
Do you ever not have service when NodeProcessManager is instantiated?

Reply:
No, but I need to pass handle to service and unfortunately two parts of this thing leaves in different worlds.